importer/tests/test_views.py

```diff
@@ -30,7 +33,53 @@ def test_upload_get(self):
     def test_upload_get_bad_repo(self):
         """GET upload page."""
         resp = self.client.get("/repositories/999/import", follow=True)
+        self.assertTrue(resp.status_code == NOT_FOUND)
+
+    def test_upload_with_permissions(self):
+        """GET upload page with different user permissions"""
+        self.logout()
+        self.login(self.USERNAME_NO_REPO)
+        # user has no permissions at all
+        resp = self.client.get("/repositories/{0}/import".format(
+            self.repo.slug), follow=True)
         self.assertTrue(resp.status_code == UNAUTHORIZED)
+        # user has author permissions and cannot see the import for the repo
+        assign_user_to_repo_group(
+            self.user_norepo,
+            self.repo,
+            GroupTypes.repo_author
+        )
+        resp = self.client.get("/repositories/{0}/import".format(
+            self.repo.slug), follow=True)
+        self.assertTrue(resp.status_code == UNAUTHORIZED)
+        # user has curator permissions and can see the the import for the repo
+        remove_user_from_repo_group(
+            self.user_norepo,
+            self.repo,
+            GroupTypes.repo_author
+        )
+        assign_user_to_repo_group(
+            self.user_norepo,
+            self.repo,
+            GroupTypes.repo_curator
+        )
+        resp = self.client.get("/repositories/{0}/import".format(
+            self.repo.slug), follow=True)
+        self.assertTrue(resp.status_code == HTTP_OK)
+        # user has admin permissions and can see the the import for the repo
+        remove_user_from_repo_group(
+            self.user_norepo,
+            self.repo,
+            GroupTypes.repo_curator
+        )
+        assign_user_to_repo_group(
+            self.user_norepo,
+            self.repo,
+            GroupTypes.repo_administrator
+        )
+        resp = self.client.get("/repositories/{0}/import".format(
```

Comment:
It would be nice to see it 403 again between `HTTP_OK` as curator and then `HTTP_OK` as admin, just so we know something actually changed.

Since that's being done so many times already in this test, maybe create a local function to do it.

```python
def assert_status_code(url, code):
    """Returns status code of GET attempt"""
    resp self.client.get(url, follow=True)
    self.assertTrue(resp.status_code == code)
```